### DISCUSSION TITLE
How can I disable the "explain" and "ask" pop-ups whenever any text is selected?

### DISCUSSION THREAD

**Discussion by murtaza-nasir** [open-webui/open-webui#8032] [Q&A]
I do not like these pop ups as I am often working with code or other text that I am copying and these keep popping up and keep getting clicked accidentally which is extremely frustrating. Is there a way to disable these?

**Comment by AapoTahkola** [comment]
This is also interfering with linux selection-clipboard copy, I hate it because I cannot middle-click paste stuff.

**Comment by AapoTahkola** [comment]
You can get rid of these by using ublock origin picker and carefully aiming it the floating panel and blocking. So happy I found a way. My opinion about these floating things that automatically appear is that this feature should be removed completely as they serve no practical purpose. What is this, language learning app I do think the assumption is that people are literate.

**Comment by shayanPRESTIGE** [comment]
These popups are a single most frustrating part of of my workflow as they cause a lot of unintended API calls that cost me a lot at times.

**Comment by FrankenLab** [comment]
I agree these are one of the most frustrating parts of using Open-WebUI. Here's my solution using Tampermonkey. For those that don't know, Tampermonkey is a browser extension (both Firefox and Chrome based browsers), that allows you to install, manage, and run custom JavaScript scripts on websites. It modifies web pages or browser behaviors by injecting user-written code at runtime.

```// ==UserScript==
// @name         OpenWebUI Disable Floating Ask/Explain Buttons
// @namespace    http://tampermonkey.net/
// @version      1.0
// @description  Disable and hide Ask/Explain buttons in OpenWebUI
// @match        *://*/*
// @grant        none
// ==/UserScript==

(function() {
    'use strict';

    function disableElement() {
        const el = document.getElementById('floating-buttons-199422b2-f258-42bf-8c69-e6803e6cefe0');
        if (el) {
            // Hide the element
            el.style.display = 'none';

            // Disable pointer events so clicks do nothing
            el.style.pointerEvents = 'none';

            // Remove from DOM completely
            el.remove();

            // Stop further attempts to disable
            return true;
        }
        return false;
    }

    // Try once immediately
    if (!disableElement()) {
        // If element doesn't exist yet, observe DOM changes and disable when it appears
        const observer = new MutationObserver(() => {
            if (disableElement()) {
                observer.disconnect();
            }
        });
        observer.observe(document.body, { childList: true, subtree: true });
    }
})();```

**Reply by murtaza-nasir** [reply]
Thank you for sharing. It is astonishing that the devs, after all this time, couldn't add a simple toggle in the settings somewhere to disable this.

**Reply by rgaricano** [reply]
Open-WebUI isn't a project of a bunch of retired devs who collect a pension and have nothing else to do but deal with the constant complaints and requests for free because my button doesn't exist... it's a collaborative effort. The surprising thing is that the devs still have the patience to address requests and personal tastes with this type of criticism and ways of asking for it.

**Reply by murtaza-nasir** [reply]
If the devs had the time to add the feature, they also have the time to add a toggle for it. I am not saying that these are not hard-working people, but just because some people work hard does not mean that other people cannot criticize the direction they take or the philosophical choices they make.

**Reply by rgaricano** [reply]
This feature was implemented a year ago, so perhaps neither the developer nor others thought back then that there would be people who would want to disable it.
With that logic, everything implemented should have a configuration, every panel, window, or function, which would mean exponential growth in code. Not only would the code be duplicated, but it would also be completely unusable.
I'm not saying it's not necessary; what I'm saying is that something so simple is easier to do by looking at the code and making a proposal that, even if it's not integrated into the official version, anyone can implement in their personal development. It's much simpler than having to implement a feature with many more lines of code and having another layer in each browser to solve it.

**Reply by murtaza-nasir** [reply]
I opened an issue for this fairly soon after this feature was released and the issue got deleted and I was told that it needs to be put here. It's been sitting here like this since then. Looking at the massively complicated user/admin settings pages of this application, can you tell me with a straight face that adding a toggle for this would add massive complication for anyone? How is a hack like the above better than having a simple toggle in the app? The hack is better than nothing. You are being obtuse and resorting to logical fallacies to defend a poor user experience and to incorrectly assert that there can be no criticism of a free product.

**Reply by rgaricano** [reply]
Murtaza, I'm sorry if I gave you that impression, but I'm by no means against criticism, especially if it's constructive, and the truth is I hadn't noticed the dates.
Anyway, if it were something that bothered users too much and "should be removed" (as has also been mentioned), I imagine there would be more insistence from the rest of us who use it and those of us who don't find it so bothersome... which suggests it's not essential.
But don't worry, even though it's not necessary for me, I'm going to prepare a PR for it, and for your sake. That doesn't mean it will be accepted. I'm just a user who contributes and doesn't make any decisions.

**Reply by murtaza-nasir** [reply]
Thank you for the thoughtful reply and your understanding. I recognize that developers on a project like this can get overwhelmed with requests. I would have attempted to work on a PR for this myself, but the project's complexity is a bit beyond my current expertise as a hobbyist.

I and a lot of other people who find the buttons disruptive would be very grateful if you prepare a PR for this. Thank you very much for offering to take it on.

**Reply by rgaricano** [reply]
you can disable it from user settings > interface > quick actions

**Reply by get-loose** [reply]
Thanks so much @rgaricano

**Reply by rgaricano** [reply]
I didn't anything on that!, I also joint myself to thanks to Team!
